skipruntime-ts/core/src/skipruntime_api.ts

```diff
@@ -243,19 +243,22 @@ export interface CollectionReader<K extends TJSON, V extends TJSON> {
   getAll(): Entry<K, V>[];
 
   /**
-   * Get a diff of a collection according a from watermark, if one exists.
-   * @param from The from watermark to retrieve diff from
-   * @returns {Watermarked} an array en key, values pair with the corresponding watermark.
+   * Get a diff of added/updated entries in a collection since some watermark.
+   * @param since A watermark from which to compute the diff
+   * @returns {Watermarked} An object containing updated entries and a new watermark string.
    */
-  getDiff(from: string): Watermarked<K, V>;
+  getDiff(since: string): Watermarked<K, V>;
```

Comment:
A question that occurs to me thinking as a user is: How are deletions expressed?

Reply:
Deletes are expressed as the key having an empty array as value.

Reply:
Ok, but my point is that to begin to see that a user would have to unpack the definition of Watermarked, and then of Entry, to see that Entry is `[K, V[]]` and then guess that a key associated to an empty array is how a delete is represented. It would probably be good to mention here on `getDiff` that removals are also represented by the `Watermarked` value, without saying how here, just leaving that to the docs of `Watermarked`. Sorry, this should be just a not to myself, I'm happy to iterate on this once this lands.

Reply:
Oh I see. Yes, you are right, this should be made explicit in the docstring. Also, I'm not a 100% convinced about having `getDiff()` return a `Watermarked<K, V>` instead of explicitly returning a `[Entry<K, V>[], boolean, Watermark]`.

Reply:
I don't know about the name `Watermarked`, but a good thing about having a type definition is that there is one canonical place to document it.

Reply:
I agree in general but in this case it seems `Watermarked` is only ever used as the return type of `getDiff()`, which hints to me that the right place to document it is in `getDiff()`.

Reply:
I've got some more changes I'll send soon renaming `Watermarked` to `CollectionUpdate` and using it also as the argument of various subscription lambdas/handles that currently take its same fields as 3 destructured arguments.  Will leave be for now on this PR.